lib/gc/Transforms/DeepTileContractionNamedOp.cpp

```diff
@@ -0,0 +1,1032 @@
+              currentOp = *result;
+            }
+          }
+        }
+      } else {
+        TilingInterface tilingInterface =
+            cast<TilingInterface>(currentOp.getOperation());
+        FailureOr<linalg::ForallTilingResult> tilingResult =
+            linalg::tileToForallOpUsingTileSizes(b, tilingInterface, tileSizes,
+                                                 std::nullopt);
+        if (failed(tilingResult))
+          return failure();
+        b.replaceOp(currentOp, tilingResult->tileOp);
+        currentOp = dyn_cast<linalg::LinalgOp>(tilingResult->tiledOp);
+      }
+    }
+  }
+  result.tiledOps.emplace_back(currentOp);
+  return result;
+}
+
+// Turn a OpFoldResult into a Value
+static Value turnOpFoldResultIntoValue(RewriterBase &rewriter, Location loc,
+                                       OpFoldResult result) {
+  if (auto value = dyn_cast<Value>(result))
+    return value;
+  if (auto attr = dyn_cast<Attribute>(result)) {
+    if (auto val = dyn_cast<IntegerAttr>(attr)) {
+      if (val.getType().isIndex())
+        return rewriter.create<arith::ConstantIndexOp>(loc, val.getInt());
+      else
+        return rewriter.create<arith::ConstantIntOp>(loc, val.getInt(),
+                                                     val.getType());
+    }
+  }
+  return Value();
+}
+
+/*
+matmul(A, B) -> C
+---------------->
+forall([PM, PN, PK]: [MThreads, NThreads, KThreads]) {
+    CSlice = [KThreads, PM * MOuterBlock: (PM + 1) * MOuterBlock,
+     PN * NOuterBlock: (PN + 1) * NOuterBlock]
+    ASlice = A[PM * MOuterBlock: (PM + 1) * MOuterBlock, PK * KOuterBlock * (PK
++ 1) * KOuterBlock]
+    BSlice = B[PK * KOuterBlock * (PK + 1) * KOuterBlock, PN *
+NOuterBlock: (PN + 1) * NOuterBlock] CSlice2 = CSlice[PK, PM * MOuterBlock: (PM
++ 1) * MOuterBlock, PN * NOuterBlock: (PN + 1) * NOuterBlock]
+    MNumBlock = MOuterBlock / MBlock
+    NNumBlock = NOuterBlock / NBlock
+    KNumBlock = KOuterBlock / KBlovk
+    for([om, on, ok]: [MNumBlock, NNumBlock, KNumBlock]) {
+      ASlice2 = ASlice[om * MBlock: (om + 1) * MBlock, ok * KBlock: (ok + 1) *
+KBlock]
+      BSlice2 = BSlice[0, om * MBlock: (om + 1) * MBlock, ok * KBlock: (ok +
+1) * KBlock]
+      CSlice3 = CSlice2[0, om * MBlock: (om + 1) * MBlock, on * NBlock:
+(on + 1) * NBlock] (init with 0 when ok == 0)
+      MNumInnerBlock = MBlock / iim_block_
+      ...
+      for([im, in]: [MNumInnerBlock, NNumInnerBlock]) {
+        ASlice3 = ASlice2[im * iim_block_: (im + 1) * iim_block_, :]
+        BSlice3 = BSlice2[0, im * iim_block_: (im + 1) * iim_block_, :]
+        CSlice4 = CSlice3[0, im * iim_block_: (im + 1) * iim_block_, in *
+iin_block_: (in + 1) * iin_block_] (init with 0 when ok == 0)
+        brgemm(bs=KNumInnerBlock, M=iim_block_, N=iin_block_, K=iik_block,
+A=ASlice3, B=BSlice3, C=CSlice4, onlyUpdate=(ok!=0));
+      }
+    }
+}
+C = final_reduce(CSlice)
+*/
+struct deepTileMatmul : public OpInterfaceRewritePattern<linalg::LinalgOp> {
```

Comment:
It is better to change the name of `struct / class` into `DeepTileMatmul` style to align with the llvm coding style.

Reply:
Thanks for the advice! The name has been changed